### DISCUSSION TITLE
Streaming ISR

### DISCUSSION THREAD

**Discussion by Milo123459** [vercel/next.js#34497] [Ideas]
### Describe the feature you'd like to request

Recently with the release of [Next.js V12.1](https://nextjs.org/blog/next-12-1) came along on-demand ISR updating. It would be very nice if there was a streaming-based approach to ISR, meaning whenever the content is invalidated, the components using the data from the `getStaticProps` automatically re-render with the new data.

### Describe the solution you'd like

Ideally, no code changes should be required. This could be made a config option if need be. 

I see this being very useful in various use cases, for example, a Github Issues Client. My app currently uses ISR and now updates on-demand, but would be really nice if the component that renders the comments would update whenever the data is invalidated so that it is updated in real-time.

### Describe alternatives you've considered

Just refresh the page, however, if you refresh the page while editing text, you will lose that progress. This is a very flawed workaround.

**Comment by Milo123459** [comment]
Has anyone got any ideas for this?

**Reply by mi-hol** [reply]
maybe https://nextjs.org/blog/next-12-2#on-demand-incremental-static-regeneration-stable is what you are looking for?

**Reply by Milo123459** [reply]
This was announced in V12.1, but has now become stable. This feature request is based around that feature.

**Comment by taylor-lindores-reeves** [comment]
Essentially a realtime version of On-Demand ISR? I agree.

**Reply by Milo123459** [reply]
Yes, so when a route is revalidated automatically get the changes and update them on the current page.

**Reply by taylor-lindores-reeves** [reply]
I'm trying to find out how to do this. Seems like there isn't a working solution at the moment. I'm considering some sort of page request on interval in combination with ISR. I actually thought ISR would update automatically. At the moment, I have ISR set up and when I update a record in the database, hit the /api/revalidate endpoint, I see the updated list of data from the database in my terminal console. However, the page itself doesn't show the updated data. I'm trying to find a solution to update the page. If you have any ideas let me know.

**Reply by Milo123459** [reply]
My entire intention with this was to avoid a fixed interval. I think websockets could work but I'm not sure he that is over-engineering.
